packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -1506,19 +1508,20 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
         const source = this.requestManager;
         if (!source) throw new Error('Request provider is not initialized!');
 
-        let request: Request | null | undefined;
-        let session: Session | undefined;
+        let requestVar: Request | null | undefined;
 
         await this._timeoutAndRetry(
             async () => {
-                request = await this._fetchNextRequest();
+                requestVar = await this._fetchNextRequest();
             },
             this.internalTimeoutMillis,
             `Fetching next request timed out after ${this.internalTimeoutMillis / 1e3} seconds.`,
         );
+        const request = requestVar; // just to make TypeScript happy
```

Comment:
Or! We could update `_timeoutAndRetry` to actually return the result value. What do you think?